Bukkit/src/main/java/com/plotsquared/bukkit/listener/PaperListener.java

```diff
@@ -407,4 +409,25 @@ public void onAsyncTabCompletion(final AsyncTabCompleteEvent event) {
         }
     }
 
+    @EventHandler
+    public void onBeaconEffect(final BeaconEffectEvent event) {
+        Block block = event.getBlock();
+        Location beaconLocation = BukkitUtil.adapt(block.getLocation());
+        Plot beaconPlot = beaconLocation.getPlot();
+        if (beaconPlot == null) {
+            return;
+        }
+
+        Player player = event.getPlayer();
+        Location playerLocation = BukkitUtil.adapt(player.getLocation());
+
+        PlotPlayer<Player> plotPlayer = BukkitUtil.adapt(player);
+        Plot playerStandingPlot = playerLocation.getPlot();
+        boolean effectDisabled = !Settings.Paper_Components.BEACON_EFFECTS ||
```

Comment:
I would not double negate the "if" so often. And the paper component check could also be implemented at the beginning. :)

Reply:
With the paper comonent check do you mean the ``if (PaperLib.isPaper()) {}`` in the ``BukkitPlatform`` class?

Reply:
https://github.com/DerEingerostete/PlotSquared/blob/d6f0abfbb40aef5fa76e4ec1d6f394bb060294d6/Bukkit/src/main/java/com/plotsquared/bukkit/listener/PaperListener.java#L414

Reply:
Thank you!

Reply:
I just tested the code and the paper component check cannot be on top because it's either disabled by config or disabled by flag. If I would put it at the beginning it would never cancel the event when set to true